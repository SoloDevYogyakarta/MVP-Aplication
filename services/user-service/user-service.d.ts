import { HttpStatus } from '@nestjs/common';
import { ChangePasswordField, LoginField, RegisterField } from '../../dto/user-dto/user-dto';
import { JwtService } from '@nestjs/jwt';
export declare class UserService {
    private readonly jwtService;
    private readonly logger;
    constructor(jwtService: JwtService);
    login(body: LoginField): Promise<{
        status: HttpStatus;
        access_token: string;
    }>;
    create(body: RegisterField): Promise<{
        status: HttpStatus;
        message: string;
        result: import("../../database/entities/authenticate/user-entity/user-entity").UserInstance;
    }>;
    update(id: number, body: RegisterField): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    changePassword(id: number, body: ChangePasswordField): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    destroy(id: number): Promise<{
        status: HttpStatus;
        message: string;
    }>;
    logout(id: number): Promise<{
        message: boolean;
        findOne: import("../../database/entities/authenticate/user-entity/user-entity").UserInstance;
    }>;
}
