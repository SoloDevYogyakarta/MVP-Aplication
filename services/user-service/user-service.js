"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
var UserService_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserService = void 0;
const common_1 = require("@nestjs/common");
const bcrypt_1 = __importDefault(require("bcrypt"));
const lodash_1 = require("lodash");
const user_entity_1 = require("../../database/entities/authenticate/user-entity/user-entity");
const environment_1 = require("../../utils/environment/environment");
const jwt_1 = require("@nestjs/jwt");
const history_entity_1 = require("../../database/entities/authenticate/history-entity/history-entity");
let UserService = UserService_1 = class UserService {
    constructor(jwtService) {
        this.jwtService = jwtService;
        this.logger = new common_1.Logger(UserService_1.name);
    }
    async login(body) {
        this.logger.log(UserService_1.name);
        const findOne = await user_entity_1.userEntity.findOne({
            where: { plat_number: body.token },
        });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Invalid username or password',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const check = await bcrypt_1.default.compareSync(body.password, findOne.password);
        if (!check) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Invalid username or password',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.is_active = true;
        findOne.save();
        const access_token = this.jwtService.sign({
            data: JSON.parse(JSON.stringify(findOne)),
        }, { secret: environment_1.environment['SECRET'] });
        return { status: common_1.HttpStatus.OK, access_token };
    }
    async create(body) {
        this.logger.log(UserService_1.name);
        const findOne = await user_entity_1.userEntity.findOne({
            where: { plat_number: body.plat_number },
        });
        if (findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Plat number already exists',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        if (body.password !== body.confirmation) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: "Password don't match, please check again",
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const create = await user_entity_1.userEntity.create((0, lodash_1.omit)({ ...body, password: await bcrypt_1.default.hashSync(body.password, 15) }, [
            'confirmation',
        ]));
        create.save();
        return {
            status: common_1.HttpStatus.CREATED,
            message: 'Account has been create',
            result: create,
        };
    }
    async update(id, body) {
        const findOne = await user_entity_1.userEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Account not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const history = await history_entity_1.userHistoryEntity.create({
            ...(0, lodash_1.omit)(JSON.parse(JSON.stringify(findOne)), [
                'id',
                'public_id',
                'role',
                'password',
                'createdAt',
                'updatedAt',
            ]),
            user_id: findOne.id,
        });
        history.save();
        findOne.update((0, lodash_1.omit)(body, ['motor', 'year_production', 'password', 'confirmation']), { where: { id } });
        return { status: common_1.HttpStatus.OK, message: 'Account has been update' };
    }
    async changePassword(id, body) {
        const findOne = await user_entity_1.userEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.INTERNAL_SERVER_ERROR,
                message: 'false',
            }, common_1.HttpStatus.INTERNAL_SERVER_ERROR);
        }
        const check = await bcrypt_1.default.compareSync(body.old_password, findOne.password);
        if (!check) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Wrong password',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        if (body.password !== body.confirmation) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: "Password don't match, please check again",
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        findOne.password = await bcrypt_1.default.hashSync(body.password, 15);
        findOne.save();
        return { status: common_1.HttpStatus.OK, message: 'Password has been update' };
    }
    async destroy(id) {
        const findOne = await user_entity_1.userEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.BAD_REQUEST,
                message: 'Account not found',
            }, common_1.HttpStatus.BAD_REQUEST);
        }
        const history = await history_entity_1.userHistoryEntity.findAll({
            where: { user_id: findOne.id },
        });
        for (const values of history) {
            values?.destroy();
        }
        findOne.destroy();
        return { status: common_1.HttpStatus.OK, message: 'Account has been delete' };
    }
    async logout(id) {
        const findOne = await user_entity_1.userEntity.findOne({ where: { id } });
        if (!findOne) {
            throw new common_1.HttpException({
                status: common_1.HttpStatus.INTERNAL_SERVER_ERROR,
                message: 'false',
            }, common_1.HttpStatus.INTERNAL_SERVER_ERROR);
        }
        findOne.is_active = false;
        findOne.save();
        return { message: true, findOne };
    }
};
exports.UserService = UserService;
exports.UserService = UserService = UserService_1 = __decorate([
    (0, common_1.Injectable)(),
    __metadata("design:paramtypes", [jwt_1.JwtService])
], UserService);
//# sourceMappingURL=user-service.js.map