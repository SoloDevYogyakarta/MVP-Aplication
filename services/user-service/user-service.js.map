{"version":3,"file":"user-service.js","sourceRoot":"","sources":["../../../src/services/user-service/user-service.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;AAAA,2CAA+E;AAC/E,oDAA4B;AAC5B,mCAA8B;AAC9B,8FAA0F;AAM1F,qEAAkE;AAClE,qCAAyC;AACzC,uGAAuG;AAGhG,IAAM,WAAW,mBAAjB,MAAM,WAAW;IAGtB,YAA6B,UAAsB;QAAtB,eAAU,GAAV,UAAU,CAAY;QAFlC,WAAM,GAAG,IAAI,eAAM,CAAC,aAAW,CAAC,IAAI,CAAC,CAAC;IAED,CAAC;IAEvD,KAAK,CAAC,KAAK,CAAC,IAAgB;QAC1B,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,aAAW,CAAC,IAAI,CAAC,CAAC;QAClC,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC;YACvC,KAAK,EAAE,EAAE,WAAW,EAAE,IAAI,CAAC,KAAK,EAAE;SACnC,CAAC,CAAC;QACH,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,8BAA8B;aACxC,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,MAAM,KAAK,GAAG,MAAM,gBAAM,CAAC,WAAW,CAAC,IAAI,CAAC,QAAQ,EAAE,OAAO,CAAC,QAAQ,CAAC,CAAC;QACxE,IAAI,CAAC,KAAK,EAAE;YACV,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,8BAA8B;aACxC,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,MAAM,YAAY,GAAG,IAAI,CAAC,UAAU,CAAC,IAAI,CACvC;YACE,IAAI,EAAE,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,OAAO,CAAC,CAAC;SAC1C,EACD,EAAE,MAAM,EAAE,yBAAW,CAAC,QAAQ,CAAC,EAAE,CAClC,CAAC;QACF,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,YAAY,EAAE,CAAC;IACjD,CAAC;IAED,KAAK,CAAC,MAAM,CAAC,IAAmB;QAC9B,IAAI,CAAC,MAAM,CAAC,GAAG,CAAC,aAAW,CAAC,IAAI,CAAC,CAAC;QAClC,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC;YACvC,KAAK,EAAE,EAAE,WAAW,EAAE,IAAI,CAAC,WAAW,EAAE;SACzC,CAAC,CAAC;QACH,IAAI,OAAO,EAAE;YACX,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,4BAA4B;aACtC,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,IAAI,IAAI,CAAC,QAAQ,KAAK,IAAI,CAAC,YAAY,EAAE;YACvC,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,0CAA0C;aACpD,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,MAAM,MAAM,GAAG,MAAM,wBAAU,CAAC,MAAM,CACpC,IAAA,aAAI,EAAC,EAAE,GAAG,IAAI,EAAE,QAAQ,EAAE,MAAM,gBAAM,CAAC,QAAQ,CAAC,IAAI,CAAC,QAAQ,EAAE,EAAE,CAAC,EAAE,EAAE;YACpE,cAAc;SACf,CAAC,CACH,CAAC;QACF,MAAM,CAAC,IAAI,EAAE,CAAC;QACd,OAAO;YACL,MAAM,EAAE,mBAAU,CAAC,OAAO;YAC1B,OAAO,EAAE,yBAAyB;YAClC,MAAM,EAAE,MAAM;SACf,CAAC;IACJ,CAAC;IAED,KAAK,CAAC,MAAM,CAAC,EAAU,EAAE,IAAmB;QAC1C,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC5D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,mBAAmB;aAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QAED,MAAM,OAAO,GAAG,MAAM,kCAAiB,CAAC,MAAM,CAAC;YAC7C,GAAG,IAAA,aAAI,EAAC,IAAI,CAAC,KAAK,CAAC,IAAI,CAAC,SAAS,CAAC,OAAO,CAAC,CAAC,EAAE;gBAC3C,IAAI;gBACJ,WAAW;gBACX,MAAM;gBACN,UAAU;gBACV,WAAW;gBACX,WAAW;aACZ,CAAC;YACF,OAAO,EAAE,OAAO,CAAC,EAAE;SACpB,CAAC,CAAC;QACH,OAAO,CAAC,IAAI,EAAE,CAAC;QACf,OAAO,CAAC,MAAM,CACZ,IAAA,aAAI,EAAC,IAAI,EAAE,CAAC,OAAO,EAAE,iBAAiB,EAAE,UAAU,EAAE,cAAc,CAAC,CAAC,EACpE,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAClB,CAAC;QACF,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,yBAAyB,EAAE,CAAC;IACvE,CAAC;IAED,KAAK,CAAC,cAAc,CAAC,EAAU,EAAE,IAAyB;QACxD,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC5D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,qBAAqB;gBACxC,OAAO,EAAE,OAAO;aACjB,EACD,mBAAU,CAAC,qBAAqB,CACjC,CAAC;SACH;QACD,MAAM,KAAK,GAAG,MAAM,gBAAM,CAAC,WAAW,CAAC,IAAI,CAAC,YAAY,EAAE,OAAO,CAAC,QAAQ,CAAC,CAAC;QAC5E,IAAI,CAAC,KAAK,EAAE;YACV,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,gBAAgB;aAC1B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,IAAI,IAAI,CAAC,QAAQ,KAAK,IAAI,CAAC,YAAY,EAAE;YACvC,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,0CAA0C;aACpD,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QACD,OAAO,CAAC,QAAQ,GAAG,MAAM,gBAAM,CAAC,QAAQ,CAAC,IAAI,CAAC,QAAQ,EAAE,EAAE,CAAC,CAAC;QAC5D,OAAO,CAAC,IAAI,EAAE,CAAC;QACf,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,0BAA0B,EAAE,CAAC;IACxE,CAAC;IAED,KAAK,CAAC,OAAO,CAAC,EAAU;QACtB,MAAM,OAAO,GAAG,MAAM,wBAAU,CAAC,OAAO,CAAC,EAAE,KAAK,EAAE,EAAE,EAAE,EAAE,EAAE,CAAC,CAAC;QAC5D,IAAI,CAAC,OAAO,EAAE;YACZ,MAAM,IAAI,sBAAa,CACrB;gBACE,MAAM,EAAE,mBAAU,CAAC,WAAW;gBAC9B,OAAO,EAAE,mBAAmB;aAC7B,EACD,mBAAU,CAAC,WAAW,CACvB,CAAC;SACH;QAED,MAAM,OAAO,GAAG,MAAM,kCAAiB,CAAC,OAAO,CAAC;YAC9C,KAAK,EAAE,EAAE,OAAO,EAAE,OAAO,CAAC,EAAE,EAAE;SAC/B,CAAC,CAAC;QAEH,KAAK,MAAM,MAAM,IAAI,OAAO,EAAE;YAC5B,MAAM,EAAE,OAAO,EAAE,CAAC;SACnB;QACD,OAAO,CAAC,OAAO,EAAE,CAAC;QAClB,OAAO,EAAE,MAAM,EAAE,mBAAU,CAAC,EAAE,EAAE,OAAO,EAAE,yBAAyB,EAAE,CAAC;IACvE,CAAC;CACF,CAAA;AAlKY,kCAAW;sBAAX,WAAW;IADvB,IAAA,mBAAU,GAAE;qCAI8B,gBAAU;GAHxC,WAAW,CAkKvB"}