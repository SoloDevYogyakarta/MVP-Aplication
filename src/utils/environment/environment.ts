import { config } from 'dotenv';

config();

export const environment: NodeJS.ProcessEnv = process.env;
