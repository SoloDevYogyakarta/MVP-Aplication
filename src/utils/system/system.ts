import fs from 'fs';
import { join } from 'path';

function joinpath(path: string): string {
  return join(__dirname, path);
}

function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

function createpath(path: string, data: unknown): void {
  let values = data;
  if (typeof values !== 'string') {
    values = JSON.stringify(values);
  }
  fs.writeFileSync(joinpath(path), values as string);
}

function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}

export { joinpath, readpath, createpath, removepath };
