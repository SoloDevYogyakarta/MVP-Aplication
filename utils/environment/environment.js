"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.environment = void 0;
const dotenv_1 = require("dotenv");
(0, dotenv_1.config)();
exports.environment = process.env;
//# sourceMappingURL=environment.js.map