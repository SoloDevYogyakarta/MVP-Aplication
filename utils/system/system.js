"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.removepath = exports.readpath = exports.createpath = exports.joinpath = void 0;
const fs_1 = __importDefault(require("fs"));
const path_1 = require("path");
function joinpath(path) {
    return (0, path_1.join)(__dirname, path);
}
exports.joinpath = joinpath;
function createpath(path, data) {
    if (typeof data !== 'string') {
        data = JSON.stringify(data);
    }
    fs_1.default.writeFileSync(joinpath(path), data);
}
exports.createpath = createpath;
function readpath(path) {
    return fs_1.default.readFileSync(joinpath(path), { encoding: 'utf-8' });
}
exports.readpath = readpath;
function removepath(path) {
    fs_1.default.unlinkSync(joinpath(path));
}
exports.removepath = removepath;
//# sourceMappingURL=system.js.map