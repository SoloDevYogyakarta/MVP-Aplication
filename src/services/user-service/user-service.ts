import { HttpException, HttpStatus, Injectable, Logger } from '@nestjs/common';
import bcrypt from 'bcrypt';
import { omit } from 'lodash';
import { userEntity } from '../../database/entities/authenticate/user-entity/user-entity';
import {
  ChangePasswordField,
  LoginField,
  RegisterField,
} from '../../dto/user-dto/user-dto';
import { environment } from '../../utils/environment/environment';
import { JwtService } from '@nestjs/jwt';

@Injectable()
export class UserService {
  private readonly logger = new Logger(UserService.name);

  constructor(private readonly jwtService: JwtService) {}

  async login(body: LoginField) {
    this.logger.log(UserService.name);
    const findOne = await userEntity.findOne({
      where: { plat_number: body.token },
    });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Invalid username or password',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    const check = await bcrypt.compareSync(body.password, findOne.password);
    if (!check) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Invalid username or password',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    const access_token = this.jwtService.sign(
      {
        data: JSON.parse(JSON.stringify(findOne)),
      },
      { secret: environment['SECRET'] },
    );
    return { status: HttpStatus.OK, access_token };
  }

  async create(body: RegisterField) {
    this.logger.log(UserService.name);
    const findOne = await userEntity.findOne({
      where: { plat_number: body.plat_number },
    });
    if (findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Plat number already exists',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    if (body.password !== body.confirmation) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: "Password don't match, please check again",
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    const create = await userEntity.create(
      omit({ ...body, password: await bcrypt.hashSync(body.password, 15) }, [
        'confirmation',
      ]),
    );
    create.save();
    return {
      status: HttpStatus.CREATED,
      message: 'Account has been create',
      result: create,
    };
  }

  async update(public_id: string, body: RegisterField) {
    const findOne = await userEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Account not found',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.update(
      omit(body, ['motor', 'year_production', 'password', 'confirmation']),
      { where: { public_id } },
    );
    return { status: HttpStatus.OK, message: 'Account has been update' };
  }

  async changePassword(public_id: string, body: ChangePasswordField) {
    const findOne = await userEntity.findOne({ where: { public_id } });
    if (!findOne) {
      throw new HttpException(
        {
          status: HttpStatus.INTERNAL_SERVER_ERROR,
          message: 'false',
        },
        HttpStatus.INTERNAL_SERVER_ERROR,
      );
    }
    const check = await bcrypt.compareSync(body.old_password, findOne.password);
    if (!check) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: 'Wrong password',
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    if (body.password !== body.confirmation) {
      throw new HttpException(
        {
          status: HttpStatus.BAD_REQUEST,
          message: "Password don't match, please check again",
        },
        HttpStatus.BAD_REQUEST,
      );
    }
    findOne.password = await bcrypt.hashSync(body.password, 15);
    findOne.save();
    return { status: HttpStatus.OK, message: 'Password has been update' };
  }
}
