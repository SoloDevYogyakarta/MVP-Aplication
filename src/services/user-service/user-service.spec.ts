import { faker } from '@faker-js/faker';
import { HttpStatus } from '@nestjs/common';
import { JwtService } from '@nestjs/jwt';
import { omit } from 'lodash';
import { createpath } from '../../utils/system/system';
import { environment } from '../../utils/environment/environment';
import { UserService } from './user-service';

describe('UserService', () => {
  let service: UserService;

  beforeEach(() => {
    const jwtService = new JwtService({
      secret: environment['SECRET'],
      signOptions: {
        expiresIn: '7d',
      },
    });
    service = new UserService(jwtService);
  });

  it('should to be defined', () => expect(service).toBeDefined());

  it('render correctly', () => expect(service).toMatchSnapshot());

  it('create', async () => {
    const result = await service.create({
      plat_number: `BG ${Math.floor(1000 + Math.random() * 9999)} JAK`,
      phone_number: faker.phone.number(),
      name: faker.person.fullName(),
      motor: faker.commerce.productName(),
      year_production: Math.floor(1000 + Math.random() * 2023),
      address: faker.location.streetAddress(),
      password: 'password',
      confirmation: 'password',
    });
    createpath('../folder-text/user-http-entity.txt', result.result);
    expect(omit(result, ['result'])).toEqual({
      status: HttpStatus.CREATED,
      message: 'Account has been create',
    });
  });
});
