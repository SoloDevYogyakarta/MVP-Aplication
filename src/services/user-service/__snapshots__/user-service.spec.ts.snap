// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService render correctly 1`] = `
UserService {
  "jwtService": JwtService {
    "logger": Logger {
      "context": "JwtService",
      "options": {},
    },
    "options": {
      "secret": "secret",
      "signOptions": {
        "expiresIn": "7d",
      },
    },
  },
  "logger": Logger {
    "context": "UserService",
    "options": {},
  },
}
`;
